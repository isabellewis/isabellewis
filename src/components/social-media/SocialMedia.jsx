import React from 'react';
import {
  SocialMediaWrapper,
  Logo,
  LogoA
} from './SocialMedia.style';

export const SocialMedia = ({
	theme
}) => (
	<SocialMediaWrapper className={`${theme}LogoWrapper`}>
		<LogoA href='https://github.com/isabellewis' target="_blank" className={`${theme}LogoA`}>
			<Logo className={`fa fa-github-alt ${theme}Logo`} />
		</LogoA>
		<LogoA href='https://www.linkedin.com/in/isabel-lewis-693779139/' target="_blank" className={`${theme}LogoA`}>
			<Logo className={`fa fa-linkedin ${theme}Logo`} />
		</LogoA>
		<LogoA href='https://twitter.com/irlewis' target="_blank" className={`${theme}LogoA`}>
			<Logo className={`fa fa-twitter ${theme}Logo`} />
		</LogoA>
		<LogoA href='https://www.instagram.com/isabel_lewis/' target="_blank" className={`${theme}LogoA`}>
			<Logo className={`fa fa-instagram ${theme}Logo`} />
		</LogoA>
	</SocialMediaWrapper>
);