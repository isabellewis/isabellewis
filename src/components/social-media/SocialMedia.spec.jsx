import React from 'react';
import { shallow } from 'enzyme';
import { SocialMedia } from './SocialMedia';


describe('SocialMedia', () => {
	let component;

	beforeEach(() => {
		component = shallow(<SocialMedia theme="footer" />);
	});

	it('should match the snapshot for footer', () => {
		expect(component).toMatchSnapshot();
	});

	it('should match the snapshot for header', () => {
    component = shallow(<SocialMedia theme="header" />);
    expect(component).toMatchSnapshot();
	});
});