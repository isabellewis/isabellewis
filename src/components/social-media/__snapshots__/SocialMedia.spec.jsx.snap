// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMedia should match the snapshot for footer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialMedia
    theme="footer"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.a
          className="footerLogoA"
          href="https://github.com/isabellewis"
          target="_blank"
        >
          <styled.i
            className="fa fa-github-alt footerLogo"
          />
        </styled.a>,
        <styled.a
          className="footerLogoA"
          href="https://www.linkedin.com/in/isabel-lewis-693779139/"
          target="_blank"
        >
          <styled.i
            className="fa fa-linkedin footerLogo"
          />
        </styled.a>,
        <styled.a
          className="footerLogoA"
          href="https://twitter.com/irlewis"
          target="_blank"
        >
          <styled.i
            className="fa fa-twitter footerLogo"
          />
        </styled.a>,
        <styled.a
          className="footerLogoA"
          href="https://www.instagram.com/isabel_lewis/"
          target="_blank"
        >
          <styled.i
            className="fa fa-instagram footerLogo"
          />
        </styled.a>,
      ],
      "className": "footerLogoWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-github-alt footerLogo"
          />,
          "className": "footerLogoA",
          "href": "https://github.com/isabellewis",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-github-alt footerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-linkedin footerLogo"
          />,
          "className": "footerLogoA",
          "href": "https://www.linkedin.com/in/isabel-lewis-693779139/",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-linkedin footerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-twitter footerLogo"
          />,
          "className": "footerLogoA",
          "href": "https://twitter.com/irlewis",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-twitter footerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-instagram footerLogo"
          />,
          "className": "footerLogoA",
          "href": "https://www.instagram.com/isabel_lewis/",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-instagram footerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.a
            className="footerLogoA"
            href="https://github.com/isabellewis"
            target="_blank"
          >
            <styled.i
              className="fa fa-github-alt footerLogo"
            />
          </styled.a>,
          <styled.a
            className="footerLogoA"
            href="https://www.linkedin.com/in/isabel-lewis-693779139/"
            target="_blank"
          >
            <styled.i
              className="fa fa-linkedin footerLogo"
            />
          </styled.a>,
          <styled.a
            className="footerLogoA"
            href="https://twitter.com/irlewis"
            target="_blank"
          >
            <styled.i
              className="fa fa-twitter footerLogo"
            />
          </styled.a>,
          <styled.a
            className="footerLogoA"
            href="https://www.instagram.com/isabel_lewis/"
            target="_blank"
          >
            <styled.i
              className="fa fa-instagram footerLogo"
            />
          </styled.a>,
        ],
        "className": "footerLogoWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-github-alt footerLogo"
            />,
            "className": "footerLogoA",
            "href": "https://github.com/isabellewis",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-github-alt footerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-linkedin footerLogo"
            />,
            "className": "footerLogoA",
            "href": "https://www.linkedin.com/in/isabel-lewis-693779139/",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-linkedin footerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-twitter footerLogo"
            />,
            "className": "footerLogoA",
            "href": "https://twitter.com/irlewis",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-twitter footerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-instagram footerLogo"
            />,
            "className": "footerLogoA",
            "href": "https://www.instagram.com/isabel_lewis/",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-instagram footerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialMedia should match the snapshot for header 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialMedia
    theme="header"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.a
          className="headerLogoA"
          href="https://github.com/isabellewis"
          target="_blank"
        >
          <styled.i
            className="fa fa-github-alt headerLogo"
          />
        </styled.a>,
        <styled.a
          className="headerLogoA"
          href="https://www.linkedin.com/in/isabel-lewis-693779139/"
          target="_blank"
        >
          <styled.i
            className="fa fa-linkedin headerLogo"
          />
        </styled.a>,
        <styled.a
          className="headerLogoA"
          href="https://twitter.com/irlewis"
          target="_blank"
        >
          <styled.i
            className="fa fa-twitter headerLogo"
          />
        </styled.a>,
        <styled.a
          className="headerLogoA"
          href="https://www.instagram.com/isabel_lewis/"
          target="_blank"
        >
          <styled.i
            className="fa fa-instagram headerLogo"
          />
        </styled.a>,
      ],
      "className": "headerLogoWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-github-alt headerLogo"
          />,
          "className": "headerLogoA",
          "href": "https://github.com/isabellewis",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-github-alt headerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-linkedin headerLogo"
          />,
          "className": "headerLogoA",
          "href": "https://www.linkedin.com/in/isabel-lewis-693779139/",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-linkedin headerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-twitter headerLogo"
          />,
          "className": "headerLogoA",
          "href": "https://twitter.com/irlewis",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-twitter headerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.i
            className="fa fa-instagram headerLogo"
          />,
          "className": "headerLogoA",
          "href": "https://www.instagram.com/isabel_lewis/",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fa fa-instagram headerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.a
            className="headerLogoA"
            href="https://github.com/isabellewis"
            target="_blank"
          >
            <styled.i
              className="fa fa-github-alt headerLogo"
            />
          </styled.a>,
          <styled.a
            className="headerLogoA"
            href="https://www.linkedin.com/in/isabel-lewis-693779139/"
            target="_blank"
          >
            <styled.i
              className="fa fa-linkedin headerLogo"
            />
          </styled.a>,
          <styled.a
            className="headerLogoA"
            href="https://twitter.com/irlewis"
            target="_blank"
          >
            <styled.i
              className="fa fa-twitter headerLogo"
            />
          </styled.a>,
          <styled.a
            className="headerLogoA"
            href="https://www.instagram.com/isabel_lewis/"
            target="_blank"
          >
            <styled.i
              className="fa fa-instagram headerLogo"
            />
          </styled.a>,
        ],
        "className": "headerLogoWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-github-alt headerLogo"
            />,
            "className": "headerLogoA",
            "href": "https://github.com/isabellewis",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-github-alt headerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-linkedin headerLogo"
            />,
            "className": "headerLogoA",
            "href": "https://www.linkedin.com/in/isabel-lewis-693779139/",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-linkedin headerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-twitter headerLogo"
            />,
            "className": "headerLogoA",
            "href": "https://twitter.com/irlewis",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-twitter headerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.i
              className="fa fa-instagram headerLogo"
            />,
            "className": "headerLogoA",
            "href": "https://www.instagram.com/isabel_lewis/",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-instagram headerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
